build.gradle.kts

```diff
@@ -258,13 +258,63 @@ configure(subprojectsWithoutToolAndSmokeTestProjects) {
     apply(plugin = "ch.tutteli.gradle.plugins.dokka")
     apply(plugin = "ch.tutteli.gradle.plugins.publish")
 
-    project.extensions.getByType<ch.tutteli.gradle.plugins.publish.PublishPluginExtension>().apply {
+    extensions.getByType<ch.tutteli.gradle.plugins.publish.PublishPluginExtension>().apply {
```

Comment:
configure<PublishPluginExtension> { ...

Reply:
First of all, thanks for the review and the suggestions :slightly_smiling_face: :+1: 

Out of curiosity, why do you suggest to use `configure`? we don't use convention plugins (so far) and I don't want to magically refer to something different than the extension, so why not use it directly?

Reply:
Well, you do specify `ch.tutteli.gradle.plugins.publish.PublishPluginExtension` type here; the only possibility is that this type represents project extension.
Note: "convention plugins != project conventions", and `configure` has little to do with `convention plugins`.
With convention plugins, direct accessors should be used instead of referring to the extensions by names.

`configure<Type> { ... }` is exactly the syntax for that: https://github.com/gradle/gradle/blob/100a43bd3e0079db6225a8652cacb8784d652240/subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/ProjectExtensions.kt#L82-L83

`configure<...>` is something that suggested in Gradle documentation for the cases when accessors are not available: https://docs.gradle.org/current/userguide/kotlin_dsl.html#sec:kotlin_using_standard_api

`extensions.getByType...apply.{` is pure noise.

Reply:
> the only possibility is that this type represents project extension.

exactly. if you use `configure<...>` then the project convention (introduced via a plugin such as java-library) does not necessarily need to load the extension but could replace it by something custom. I don't like such magic things.

But I see that conventions are already deprecated at will be removed with 9.0.0. My guess is, that after the removal, this will always only load the extension. Therefore it does then the same thing and I agree that it is more gradle like to use `configure`, so I am going to rewrite it